src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService.cs

```diff
@@ -233,7 +233,7 @@ protected static T Cast<T>(object value)
                     currentDestination = member.TypeSwitch(
                         (IEventSymbol @event) => this.AddEvent(currentDestination, @event, options, availableIndices),
                         (IFieldSymbol field) => this.AddField(currentDestination, field, options, availableIndices),
-                        (IPropertySymbol property) => this.AddProperty(currentDestination, property, options, availableIndices),
+                        (IPropertySymbol property) => this.AddProperty(currentDestination, property, ShouldGenerateMethodBody(options, property), availableIndices),
```

Comment:
Why not pass the same parameters as before and instead call `ShouldGenerateMethodMody()` inside of `AddProperty`?